core/types.ml

```diff
@@ -227,8 +228,12 @@ let is_field_spec_body = is_type_body
 module Transform : TYPE_VISITOR =
 struct
   class visitor = object ((o : 'self_type))
-    val rec_vars : (meta_type_var) IntMap.t = IntMap.empty
 
+    (* (Samo) need to disable this sometimes; hacky solution *)
```

Comment:
What hack?

Reply:
The visitor sometimes needs to keep the original tyvars, and sometimes in needs to refresh them (having only one or the either broke either sugar or the unifier (I think due to decycling)).

Since I don't know exactly why the issues happened, I created this value in the class, and I'm just setting it where I need it to be different from default. It felt hacky at the time (mainly because I didn't understand what was going on), though in hindsight this is probably clean enough.